src/Joblify.Core.Api/User/UserController.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Threading.Tasks;
+using Joblify.Core.Api.Infrastructure.ActionFilterAttributes;
+using Joblify.Core.User;
+using Joblify.Core.User.Dto;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Joblify.Core.Api.User
+{
+    [Route("api/user")]
+    public class UserController : Controller
+    {
+        private readonly IUserService _userService;
+
+        public UserController(IUserService loginService)
+        {
+            _userService = loginService;
+        }
+
+        [HttpGet("{email}")]
+        public async Task<IActionResult> CheckIfEmailExists(string email)
+        {
+            var result = await _userService.CheckIfUserExists(email);
+
+            return Ok(result);
+        }
+
+        [HttpPost("saveProfile")]
+        [ValidateModel]
+        public async Task<IActionResult> SaveProfile([FromBody] EditProfileDto dto)
+        {
+            var result = await _userService.SaveProfile(dto);
+
+            if (result == null)
+            {
+                return BadRequest();
+            }
+
+            return Created("", result);
+        }
+
+        [HttpDelete("deleteProfile")]
+        public async Task<IActionResult> DeleteUser(string email)
```

Comment:
I know that we don't have time so for now it's ok, but if we want to have REST API, we should use only Get, Post, Put, Delete methods specific for User in that case. So, if we want to check if email is available there should be controller specific for that. In current situation we would have two GET methods if we wanted to return User info.

Reply:
Why? Isn't checking email (which is in our system a way of checking if _user_ exists) perfectly fine to be here? I don't think I understand fully last sentence tho, can you please elaborate?

Reply:
I fixed some parts of code below. Please tell if am I supposed to do something more with this.